.github/workflows/snap-store-publish-to-candidate.yml

```diff
@@ -78,21 +82,78 @@ jobs:
         run: |
           rev=$(snapcraft push $SNAP_NAME --release=$CHANNEL)
           echo "revision_${ARCHITECTURE}=${rev}" >> $GITHUB_OUTPUT
-    outputs: 
-      revision_amd64: ${{ steps.publish.outputs.revision_amd64 }}
-      revision_arm64: ${{ steps.publish.outputs.revision_arm64 }}
-      version: ${{ steps.build.outputs.version }}
+
+      - name: Upload snap artifact for testing
+        uses: actions/upload-artifact@v3
+        with:
+          name: snap
+          path: ${{ steps.build.outputs.snap }}
+
+  # Create an issue using the template that asks maintainers to test the snap, and take
+  # action to promote the revisions to stable if testing is successful.
   create_issue:
     name: "Create call for testing"
     needs: build
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
       - uses: JasonEtco/create-an-issue@v2
+        id: comment
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           revision_amd64: ${{ needs.build.outputs.revision_amd64 }}
           revision_arm64: ${{ needs.build.outputs.revision_arm64 }}
           version: ${{ needs.build.outputs.version }}
         with:
           filename: .github/testing-issue-template.md
+    outputs:
+      issue_number: ${{ steps.comment.outputs.number }}
+
+  # Deploy the snap inside a desktop VM and grab screenshots of the desktop, and of
+  # the application, then post them as a comment on the issue created above.
+  grab-screenshots:
+    name: Gather screenshots
+    needs: 
+      - build
+      - create_issue
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v4
+
+      - name: Get the snap artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: snap
+
+      - name: Setup LXD
+        uses: canonical/setup-lxd@v0.1.1
+
+      - name: Setup test machine and start Signal
+        id: screenshots
+        run: |
+          .github/vmctl prepare
```

Comment:
Is the file saved as executable in the repo? Not sure how to check that in a PR.

Reply:
It is indeed, and the permissions are in the git index so I don't think we need to do anything else here. It would take someone explicitly cloning the repo, removing the permissions, then committing and pushing.